x/slashing/internal/keeper/signing_info.go

```diff
@@ -77,6 +87,73 @@ func (k Keeper) IterateValidatorMissedBlockBitArray(ctx sdk.Context,
 	}
 }
 
+// Slash attempts to slash a validator. The slash is delegated to the staking
+// module to make the necessary validator changes.
+func (k Keeper) Slash(ctx sdk.Context, consAddr sdk.ConsAddress, power, distributionHeight int64) {
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeSlash,
+			sdk.NewAttribute(types.AttributeKeyAddress, consAddr.String()),
+			sdk.NewAttribute(types.AttributeKeyPower, fmt.Sprintf("%d", power)),
+			sdk.NewAttribute(types.AttributeKeyReason, types.AttributeValueDoubleSign),
+		),
+	)
+
+	fraction := k.SlashFractionDoubleSign(ctx)
```

Comment:
I know this is the case for double signing, but should we make the `Slash` function general enough so that other modules can use it too with their custom slashing fractions? Eg: tBTC and Peggy need to use the slashing logic too. 

If you disagree (which could be ok), what would be the recommended steps for them to implement slashing with their custom fractions for their use cases?

Reply:
wooaaa, great catch! This was **not** my intention and is a bug! The fraction needs to be passed to this method -- making it general.

Reply:
updated to take a fraction which the evidence module supplies.